### DISCUSSION TITLE
Keep Alive method - RTSP

### DISCUSSION THREAD

**Discussion by AdiAbuAli** [bluenviron/mediamtx#671] [Questions]
Hi,
Am using synology surveillance,  with Wyze bridge over RTSP/TCP....
For keep alive, there are 3 options in synology:
Disable 
Options (Default)
GET_Parameters

I choosed get_parameters (fixed each 5s) based on the out of Describe command response from server...

Is this right?...can I disable it?

**Comment by aler9** [comment]
You can either use OPTIONS or GET_PARAMETERS, the server accepts both and periodically checks the timing of the last keepalive request.

**Reply by AdiAbuAli** [reply]
Ok...From server side, the keep alive timer correspomf to RTSP ReadTimeOut setting?

**Reply by aler9** [reply]
I checked, and the server keepalive is currently hardcoded at 60 seconds:

https://github.com/aler9/gortsplib/blob/235cc4061beb967542b19c6be36d3ffdd8827a51/server.go#L183

therefore you have to send an OPTIONS or GET_PARAMETERS request every 50 secs.

**Comment by AdiAbuAli** [comment]
5s is hardcoded in my surveillance station...
Anyways, whats the impact of ReadTimeOut timer, if its hardcoded to 60s?

Thanks,
________________________________

**Reply by aler9** [reply]
no impact at all, the server just makes sure that a keepalive (OPTIONS or GET_PARAMETER) is received at least once every 60secs. If you send one every 5secs there are no problems at all.

[a comment by an automated account is not shown]
